src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/draftorders/judgeapproval/SortKey.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.draftorders.judgeapproval;
+
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+
+/**
+ * Represents a sorting key for judge approval draft orders, incorporating hearing time, hearing date,
+ * document submission date, and document filename. This class implements {@link Comparable} to enable
+ * sorting based on the defined properties in a priority order.
+ *
+ * <p>Sorting Priority:</p>
+ * <ol>
+ *     <li>Hearing Date (ascending, nulls first)</li>
+ *     <li>Hearing Time (ascending, nulls first)</li>
+ *     <li>Document Submitted Date (ascending, nulls first)</li>
+ *     <li>Document Filename (ascending, nulls first)</li>
+ * </ol>
+ *
+ * <p>Null-safe comparison is applied for all fields.</p>
+ *
+ * <p>Usage Example:</p>
+ * <pre>
+ * SortKey key1 = new SortKey("10:00", LocalDate.of(2024, 11, 22), LocalDateTime.of(2024, 11, 21, 9, 0), "doc1.pdf");
+ * SortKey key2 = new SortKey("11:00", LocalDate.of(2024, 11, 22), LocalDateTime.of(2024, 11, 21, 10, 0), "doc2.pdf");
+ * int result = key1.compareTo(key2);
+ * </pre>
+ *
+ * @see Comparable
+ */
+@Setter
```

Comment:
You don't need @Getter and @Setter with @Data

Reply:
[e855fa8](https://github.com/hmcts/finrem-case-orchestration-service/pull/1956/commits/e855fa84d80b254571c95fe3ae8123ad0c5b8b5c)